c10/util/Half.h

```diff
@@ -450,7 +450,7 @@ typename std::enable_if<std::is_integral<From>::value && !std::is_same<From, boo
     return f > limit::max() ||
         (f < 0 && -static_cast<uint64_t>(f) > limit::max());
   } else {
-    return f < limit::lowest() || f > limit::max();
+    return f < static_cast<From>(limit::lowest()) || f >= static_cast<From>(limit::max());
```

Comment:
This doesn't seem correct. We probably should add a unit test to validate. 

If you convert limit::max to the same type as f, f can never be greater than the maximum value representable as type From.

Reply:
> If you convert limit::max to the same type as f, f can never be greater than the maximum value representable as type From.

This is true if `To` is a wider type. I think this check is to guard against the case where `To` is a narrower type than `From`, where overflow can happen. 

Also note that implicit conversion is always happening as the warning indicated. The change is just to suppress the warning by making that conversion explicit.

> We probably should add a unit test to validate.

Alternatively, I can disable this warning for this project or file, and hand over this to pytorch developers. We folks on the LLVM compiler team aren't familiar with the testing infrastructure of pytorch, and the primary goal here to unblock internal build breaks due to this warning with latest LLVM.

Reply:
> This is true if To is a wider type. I think this check is to guard against the case where To is a narrower type than From, where overflow can happen.

Good point, but I'm still not feeling very confident. E.g. `>=` seems to suggest that the maximum value representable as From overflows in To even when From==To? It's also now inconsistent with the `f > limit::max();` in the other implementation of `overflows(From f)` below.

With signed/unsigned differences as well, I think a test would help us be sure the code is correctly checking the maximum representable values.